### ISSUE TITLE
doctl snapshot snapshot-name parameter is not respected

### ISSUE THREAD

**Issue by iosdev-republicofapps** [digitalocean/doctl#328]
Create a snapshot on a droplet with name MyDroplet and DropletID 97283253
```
doctl compute droplet-action snapshot 97283253 --snapshot-name foo-bar
```
The snapshot gets created but it gets named `MyDroplet-1528914772` even though I specified `--snapshot-name foo-bar`.  So the snapshot just gets named `DropletId-UnixTimestamp` despite my explicit instructions to name it `foo-bar`.  I also tried `-snapshot-name=foo-bar` to no avail.

I am following the `doctl` help exactly for this:
```
doctl compute droplet-action snapshot --help
snapshot droplet

Usage:
  doctl compute droplet-action snapshot <droplet-id> [flags]

Flags:
      --format string          Columns for output in a comma separated list. Possible values: ID,Status,Type,StartedAt,CompletedAt,ResourceID,ResourceType,Region
      --no-header              hide headers
      --snapshot-name string   Snapshot name (required)
      --wait                   Wait for action to complete
```

Any ideas why this isn't working?  Thanks!

**Comment by iosdev-republicofapps** [comment]
Just FYI, I had just now installed doctl from Homebrew on macOS 10.13.4 and I'm using doctl `doctl version 1.8.2-release` which is the latest version homebrew.

**Comment by JulesAU** [comment]
+1 - which means that it's impossible to conduct snapshots of volumes (i.e. non droplets):

Trying to use your official API client to automatically snapshot a volume, but it's failing because it does not include the "snapshot-name" in the HTTP POST to the endpoint:

```
./doctl version
doctl version 1.8.2-release
Git commit hash: 0b2fac0

./doctl compute volume snapshot --snapshot-name testsnapshot xxxx-xxxx-xxx-xxxx --trace
```

Fails with:

```
doctl: 2018/06/13 02:04:25 -> "POST /v2/volumes/xxxx-xxxx-xxx-xxxx/snapshots HTTP/1.1\r\nHost: api.digitalocean.com\r\nUser-Agent: doctl/1.7.2-stable godo/1.3.0\r\nContent-Length: 80\r\nAccept: application/json\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n{\"volume_id\":\"xxxx-xxxx-xxx-xxxx\",\"name\":\"\",\"description\":\"\"}\n"
create a volume snapshot

[...]

Error: POST https://api.digitalocean.com/v2/volumes/xxxx-xxxx-xxx-xxxx/snapshots: 400 param is missing or the value is empty: name
```

**Comment by iosdev-republicofapps** [comment]
It seems to be working now.  The only thing that changed is that I upgraded from doctl version 1.8.2-release to 1.8.3-release.  So I guess it was broken in 1.8.2 but is fixed in 1.8.3.  Closing.  Thanks.

Hey @JulesAU - I too was on doctl 1.8.2 and while it was broken.  Upgrading to 1.8.3 fixed the issue and snapshots are now named according to the command line arg.  Give upgrading a try - hopefully it resolves it for you also!  :-)
